boards/build_scripts/src/default.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright Tock Contributors 2024.
+
+//! Provide helpers for building Tock boards that match the default conventions.
+
+use std::fs;
+use std::path::Path;
+
+const LINKER_SCRIPT: &str = "layout.ld";
+
+/// Setup the Tock board to build with a board-provided linker script called
+/// `layout.ld`.
+///
+/// The board linker script (i.e., `layout.ld`) should end with the command:
+///
+/// ```
+/// INCLUDE tock_kernel_layout.ld
+/// ```
+///
+/// This function will ensure that the linker's search path is configured to
+/// find `tock_kernel_layout.ld`.
+pub fn default_linker_script() {
+    if !Path::new(LINKER_SCRIPT).exists() {
+        panic!("Boards must provide a `layout.ld` link script file");
+    }
+
+    // The `RUSTFLAGS` that the Tock config files set can be easily overridden
+    // by command line flags. The build will still succeed but the resulting
+    // binary may be invalid as it was not built with the intended flags. This
+    // check seeks to prevent that. Our approach is we set a sentinel flag in
+    // our configuration file and then check that it is set here. If it isn't,
+    // the flags were overwritten and the build will be invalid.
+    //
+    // We only do this check if we are actually building for an embedded target
+    // (i.e., the TARGET is not the same as the HOST). This avoids a false
+    // positive when running tools like `cargo clippy`.
+    //
+    // If you are intentionally not using the standard Tock config files, set
+    // `cfg-tock-buildflagssentinel` in your cargo config to prevent this
+    // error.
+    if std::env::var("HOST") != std::env::var("TARGET") {
+        let rust_flags = std::env::var("CARGO_ENCODED_RUSTFLAGS");
+        if !rust_flags
+            .iter()
+            .any(|f| f.contains("cfg_tock_buildflagssentinel"))
+        {
+            panic!(
+                "Incorrect build configuration. Verify you are using unstable cargo and have not unintentionally set the RUSTFLAGS environment variable."
+            );
+        }
+    }
+
+    // Include the folder where this build_script crate's Cargo.toml is in the
+    // linker file search path for `tock_kernel_layout.ld`.
+    println!("cargo:rustc-link-arg=-L{}", std::env!("CARGO_MANIFEST_DIR"));
+
+    // Include the folder where the board's Cargo.toml is in the linker file
+    // search path.
+    println!(
+        "cargo:rustc-link-arg=-L{}",
+        std::env::var("CARGO_MANIFEST_DIR").unwrap()
+    );
+    // `-Tlayout.ld`: Use the linker script `layout.ld` all boards must provide.
+    println!("cargo:rustc-link-arg=-T{}", LINKER_SCRIPT);
+
+    track_linker_script(LINKER_SCRIPT);
+}
+
+/// Track the given linker script and all of its `INCLUDE`s so that the build
+/// is rerun when any of them change.
+fn track_linker_script<P: AsRef<Path>>(path: P) {
+    let path = path.as_ref();
+
+    // Skip the default Tock linker script.
+    if path.to_str() == Some("tock_kernel_layout.ld") {
+        return;
+    }
```

Comment:
This is different from the original build.rs -- can you explain why you added this? It seems like we would want to rerun the build if that linker script is changed.